homeassistant/components/foscam/switch.py

```diff
@@ -22,63 +81,168 @@ async def async_setup_entry(
+                f"{coordinator.config_entry.entry_id}_{description.key}"
+            )
+        else:
+            return
+        if self.entity_description.key == "is_asleep":
+            self._state = self.coordinator.data["is_asleep"]["status"]
+        else:
+            self._state = self.coordinator.data.get(self.entity_description.key, False)
 
     @property
-    def is_on(self):
-        """Return true if camera is asleep."""
-        return self.is_asleep
+    def is_on(self) -> bool:
+        """Return the state of the switch."""
+        return int(self._state) != 0
 
     async def async_turn_off(self, **kwargs: Any) -> None:
-        """Wake camera."""
-        LOGGER.debug("Wake camera")
-
-        ret, _ = await self.hass.async_add_executor_job(
-            self.coordinator.session.wake_up
-        )
-
+        """Turn off the entity."""
+        if self.entity_description.key == "is_Flip":
+            ret, _ = await self.hass.async_add_executor_job(
+                self.coordinator.session.flip_video, 0
+            )
+        elif self.entity_description.key == "is_Mirror":
+            ret, _ = await self.hass.async_add_executor_job(
+                self.coordinator.session.mirror_video, 0
+            )
+        elif self.entity_description.key == "is_openIr":
+            ret, _ = await self.hass.async_add_executor_job(
+                self.coordinator.session.set_infra_led_config, 0
+            )
+            ret, _ = await self.hass.async_add_executor_job(
+                self.coordinator.session.open_infra_led
+            )
+        elif self.entity_description.key == "is_asleep":
+            ret, _ = await self.hass.async_add_executor_job(
+                self.coordinator.session.wake_up, 0
+            )
+        elif self.entity_description.key == "is_openWhiteLight":
+            ret, _ = await self.hass.async_add_executor_job(
+                self.coordinator.session.closeWhiteLight
+            )
+        elif self.entity_description.key == "is_sirenalarm":
+            ret, _ = await self.hass.async_add_executor_job(
+                self.coordinator.session.setSirenConfig, 0, 100, 0
+            )
+        elif self.entity_description.key == "is_TurnOffVolume":
+            ret, _ = await self.hass.async_add_executor_job(
+                self.coordinator.session.setVoiceEnableState, 1
+            )
+        elif self.entity_description.key == "is_TurnOffLight":
+            ret, _ = await self.hass.async_add_executor_job(
+                self.coordinator.session.setLedEnableState, 1
+            )
+        elif self.entity_description.key == "is_OpenWdr":
+            ret, _ = await self.hass.async_add_executor_job(
+                self.coordinator.session.setWdrMode, 0
+            )
+        elif self.entity_description.key == "is_OpenHdr":
+            ret, _ = await self.hass.async_add_executor_job(
+                self.coordinator.session.setHdrMode, 0
+            )
```

Comment:
Extend the SwitchEntityDescription with a turn_off_fn and make that a callable, this way you can move this method to the entity description isntead